client/rpmtrans.c

```diff
@@ -480,12 +465,57 @@ TDNFTransAddInstallPkg(
                   fp,
                   pszFilePath,
                   &rpmHeader);
-    //If not checking gpg sigs, ignore signature errors
-    if(!nGPGSigCheck && (dwError == RPMRC_NOTTRUSTED || dwError == RPMRC_NOKEY))
+
+    Fclose(fp);
+    fp = NULL;
+
+    if (dwError != RPMRC_NOTTRUSTED && dwError != RPMRC_NOKEY)
     {
-        dwError = 0;
+        BAIL_ON_TDNF_RPM_ERROR(dwError);
+    }
+    else if(nGPGSigCheck)
+    {
+        dwError = TDNFGetGPGSignatureCheck(pTdnf, pszRepoName, &nGPGSigCheck, &pszUrlGPGKey);
```

Comment:
This check is done after downloading first rpm from the transaction and user is prompted and then proceeded to download next set of dependencies. Is it possible to make this check separately after downloading all the RPMs for the transaction because the right now prompt is confusing to the user as it is in the middle of download for a single RPM but it uses the same imported key for all the RPMs in the transaction.
```
root@photon-3-dev [ ~/tdnf-keerthanak-github/tdnf ]# ./build/bin/tdnf upgrade curl

Upgrading:
curl-devel                                               x86_64                       7.61.1-8.ph3                          photon-updates                   766.02k 784405
curl-libs                                                x86_64                       7.61.1-8.ph3                          photon-updates                   528.69k 541376
curl                                                     x86_64                       7.61.1-8.ph3                          photon-updates                   228.12k 233595

Total installed size:   1.49M 1559376
Is this ok [y/N] y

Downloading:
curl-devel                              703935   100%
importing key from file:///etc/pki/rpm-gpg/VMWARE-RPM-GPG-KEY
Is this ok [y/N]:  y
curl-libs                               268559   100%
curl                                    129034   100%
Testing transaction
Running transaction
Installing/Updating: curl-libs-7.61.1-8.ph3.x86_64
Installing/Updating: curl-7.61.1-8.ph3.x86_64
Installing/Updating: curl-devel-7.61.1-8.ph3.x86_64
Removing: curl-devel-7.61.1-4.ph3.x86_64
Removing: curl-7.61.1-4.ph3.x86_64
Removing: curl-libs-7.61.1-4.ph3.x86_64

Complete!
```

Reply:
It is different from dnf or yum behavior, but I do not think it is confusing. Note that this isn't really caused by this change, signatures were already verified directly after download. It's only now apparent because of the added question.

We can change this, but this could be in a separate PR.

Reply:
Yes, I agree this is the original tdnf behavior. Since the previous code does not have the prompt, we wouldn't know when the check is done. We can take it in separate PR.